universum/main.py

```diff
@@ -41,14 +44,14 @@ def define_arguments(argument_parser):
                                           "in this case full analysis report will be published")
 
     def __init__(self, *args, **kwargs):
-        super(Main, self).__init__(*args, **kwargs)
-        self.vcs = self.vcs_factory()
-        self.launcher = self.launcher_factory()
-        self.artifacts = self.artifacts_factory()
-        self.reporter = self.reporter_factory()
-        self.code_report_collector = self.code_report_collector()
+        super().__init__(*args, **kwargs)
+        self.vcs: vcs.MainVcs = self.vcs_factory()
```

Comment:
Do we have to annotate it here?
My guess was that due to gravity being annotated mypy should be able to figure out types correctly.
Or this annotation works as double checks?

Reply:
we don't have to annotate it here, but I prefer to annotate all self.blabla variables in __init__ as that is a natural place for people to look for attribute types

Reply:
Plus double-check, as you mentioned. Even if gravity changes, we want these variables to keep their types.

Reply:
Ok, readability is enough reason to keep it.